tests/test_geoserver.py

```diff
@@ -46,17 +63,90 @@ def get_geoserver_settings():
     return geoserver_settings
 
 
+def prepare_geoserver_test_workspace(test_instance, geoserver_handler, workspace_key):
+    """
+    Prepares a workspace, its datastore and a test shapefile along with the associated Geoserver resources.
+    """
+    # Creating workspace and datastore needed to publish a shapefile
+    workspace_name = test_instance.workspaces[workspace_key]
```

Comment:
The same pre-check of `if not folder.startswith("/tmp/"):` as in the reset function should be done to ensure that files are not "successfully" created, but fail deletion at the end of the tests.